x/proof/keeper/msg_server_submit_proof_test.go

```diff
@@ -0,0 +1,1336 @@
+	sessionTree relayer.SessionTree,
+	closestProofPath []byte,
+) *types.MsgSubmitProof {
+	t.Helper()
+
+	// Generate a closest proof from the session tree using expectedClosestMerkleProofPath.
+	merkleProof, err := sessionTree.ProveClosest(closestProofPath)
+	require.NoError(t, err)
+	require.NotNil(t, merkleProof)
+
+	// Serialize the closest merkle proof.
+	merkleProofBz, err := merkleProof.Marshal()
+	require.NoError(t, err)
+
+	return &types.MsgSubmitProof{
+		SupplierAddress: supplierAddr,
+		SessionHeader:   sessionHeader,
+		Proof:           merkleProofBz,
+	}
+}
+
+func fillSessionTree(
+	ctx context.Context,
+	t *testing.T,
+	supplierKeyUid string,
+	supplierAddr string,
+	requestHeader *sessiontypes.SessionHeader,
+	responseHeader *sessiontypes.SessionHeader,
+	keyRing keyring.Keyring,
+	ringClient crypto.RingClient,
+	sessionTree relayer.SessionTree,
+	numRelays uint,
+) {
+	t.Helper()
+
+	for i := 0; i < int(numRelays); i++ {
+		idxKey := make([]byte, 64)
+		binary.PutVarint(idxKey, int64(i))
+
+		relay := newSignedEmptyRelay(
+			ctx, t,
+			supplierKeyUid,
+			supplierAddr,
+			requestHeader,
+			responseHeader,
+			keyRing,
+			ringClient,
+		)
+		relayBz, err := relay.Marshal()
+		require.NoError(t, err)
+
+		err = sessionTree.Update(idxKey, relayBz, 1)
+		require.NoError(t, err)
+	}
+}
+
+func createClaimAndStoreBlockHash(
+	ctx context.Context,
+	t *testing.T,
+	supplierAddr string,
+	appAddr string,
+	service *sharedtypes.Service,
+	sessionTree relayer.SessionTree,
+	sessionHeader *sessiontypes.SessionHeader,
+	msgServer types.MsgServer,
+	keepers *keepertest.ProofModuleKeepers,
+) {
+
+	validMerkleRootBz, err := sessionTree.Flush()
+	require.NoError(t, err)
+
+	// Create a valid claim.
+	validClaimMsg := newTestClaimMsg(t,
+		sessionHeader.GetSessionId(),
+		supplierAddr,
+		appAddr,
+		service,
+		validMerkleRootBz,
+	)
+	_, err = msgServer.CreateClaim(ctx, validClaimMsg)
+	require.NoError(t, err)
+
+	// TODO(@Red0ne) add a comment explaining why we have to do this.
+	validProofSubmissionHeight :=
+		validClaimMsg.GetSessionHeader().GetSessionEndBlockHeight() +
+			sessionkeeper.GetSessionGracePeriodBlockCount()
+
+	// Set block height to be after the session grace period.
+	validBlockHeightCtx := keepertest.SetBlockHeight(ctx, validProofSubmissionHeight)
+
+	// Set the current block hash in the session store at this block height.
+	keepers.StoreBlockHash(validBlockHeightCtx)
+}
+
+func getClosestRelayDifficultyBits(
+	t *testing.T,
+	sessionTree relayer.SessionTree,
+	closestMerkleProofPath []byte,
+) uint64 {
+	validClosestMerkleProof, err := sessionTree.ProveClosest(closestMerkleProofPath)
+	require.NoError(t, err)
+
+	validClosestMerkleProofBz, err := validClosestMerkleProof.Marshal()
+	require.NoError(t, err)
+
+	validSparseMerkleClosestProof := new(smt.SparseMerkleClosestProof)
+	err = validSparseMerkleClosestProof.Unmarshal(validClosestMerkleProofBz)
+	require.NoError(t, err)
+
+	sumSize := 8
```

Comment:
Can we retrieve this from the smt repo (no magic numbers with cross-repo dependencies)?

If not, please add TODO

Reply:
Removed the need for this.